module/zfs/zio.c

```diff
@@ -2519,8 +2518,7 @@ zio_ddt_collision(zio_t *zio, ddt_t *ddt, ddt_entry_t *dde)
 			    &aflags, &zio->io_bookmark);
 
 			if (error == 0) {
-				if (arc_buf_size(abuf) != zio->io_orig_size ||
```

Comment:
Sorry, I forgot that this check got moved up (the `if (BP_GET_LSIZE(&blk) != zio->io_orig_size)` check) outside of this diff. Nevermind!